### PR TITLE
New custom keymap for Glorious GMMK Pro ANSI layout

### PR DESCRIPTION
**Pull request by gourdo1**
## Description

This Windows-centric layout is based on Jonavin's GMMK Pro ANSI layout (https://github.com/qmk/qmk_firmware/tree/master/keyboards/gmmk/pro/ansi/keymaps/jonavin) with modifications to keymap, functionality and RGB.

Details in the readme.md:
https://github.com/gourdo1/qmk_firmware/tree/PR-branch/keyboards/gmmk/pro/ansi/keymaps/gourdo1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

none

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        }
+    }
+
+    void suspend_power_down_user(void) {
+        rgb_matrix_set_suspend_state(true);
+    }
+
+    void suspend_wakeup_init_user(void) {
+        rgb_matrix_set_suspend_state(false);
+    }
```
Comment (drashna) [inline keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c:323]:
```suggestion
```

File: keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+#endif
+
+
+void keyboard_post_init_keymap(void) {
+    // keyboard_post_init_user() moved to userspace
+    #ifdef RGB_MATRIX_ENABLE
+        rgb_matrix_mode(RGB_MATRIX_SOLID_COLOR);
```
Comment (drashna) [inline keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c:330]:
```suggestion
        rgb_matrix_mode_noeeprom(RGB_MATRIX_SOLID_COLOR);
```

File: keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        rgb_matrix_set_color_all(RGB_NAUTILUS); // Default startup colour
```
Comment (drashna) [inline keyboards/gmmk/pro/ansi/keymaps/gourdo1/keymap.c:331]:
```suggestion
        rgb_matrix_sethsv_noeeprom(129, 255, 170); // Default startup colour
```

**Review by drashna — CHANGES_REQUESTED** [review]
please remove the PDF file. 

Also, please run a formatting pass of the files in questin.

**Comment by gourdo1** [comment]
PDF file removed, all requested code changes made. All .c and .h files formatted.

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**